src/mono/mono/mini/debugger-engine.c

```diff
@@ -433,14 +402,6 @@ mono_de_set_breakpoint (MonoMethod *method, long il_offset, EventRequest *req, M
 
 	mono_loader_lock ();
 
-	CollectDomainData user_data;
-	memset (&user_data, 0, sizeof (user_data));
-	user_data.bp = bp;
-	user_data.methods = methods;
-	user_data.method_domains = method_domains;
-	user_data.method_seq_points = method_seq_points;
-	mono_de_foreach_domain (collect_domain_bp, &user_data);
```

Comment:
This doesn't seem right. Without this code, `methods->len` is always 0, so the loop below doesn't do anything.  So it seems like the whole `mono_de_set_breakpoint` doesn't do anything.  Don't you at least need to set a breakpoint on the root domain?

Reply:
You are right, I pushed my changes and then I tested, I'm sorry.
So I think we should use get_default_jit_mm everywhere.